### ISSUE TITLE
LDoc is incorrectly parsed

### ISSUE THREAD

**Issue by necauqua** [EmmyLua/IntelliJ-EmmyLua#287]
Environment
--------------
   name   |   version  
----------|----------
 IDEA version         | 2019.2.2
 EmmyLua version | 1.3.0.99-IDEA182
 OS                        | Arch Linux

If we are speaking about the same [LDoc](https://stevedonovan.github.io/ldoc/manual/doc.md.html), then current implementation in this plugin is wrong.
It marks comments as docs only if all lines are started with `---` although only the first one should start with that.

This is ok:

![image](https://user-images.githubusercontent.com/33968278/67146677-1d2cc400-f296-11e9-9e35-64d07be04ed7.png)

However this does not work, although it should (in [LDoc](https://stevedonovan.github.io/ldoc/manual/doc.md.html) manual scroll to nearest function example):

![image](https://user-images.githubusercontent.com/33968278/67146701-4e0cf900-f296-11e9-8997-8b7bba328c0e.png)

<kbd>ctrl+q</kbd> also shows no documentation for the latter case.
A big example would be [awesome wm](https://github.com/awesomeWM/awesome/blob/master/lib/naughty/core.lua#L24).

Also, when you press <kbd>Backspace</kbd> after `---` it removes all three characters.
I guess this is intentional, but please at least add a config option to disable that

**Comment by tangzx** [comment]
Sorry, emmylua does not support `LDoc`。It's emmy-style doc(`EmmyDoc`), all lines must start with `---`, the comment like `@xxx` is called `annotation`.  All annotations are listed [here](https://emmylua.github.io/annotation.html).

**Comment by necauqua** [comment]
Aha, understood
But LDoc is still widely used and if you use a well-documented library that uses LDoc then you cant see all that documentation.
I mean that even partially supporting LDoc should be easy, just allow `---` only to be the first line of the doc comment, LDoc also has annotations and most of them work the same way afaiu.

**Comment by adriweb** [comment]
(Related: https://github.com/EmmyLua/IntelliJ-EmmyLua/issues/75 and https://github.com/EmmyLua/IntelliJ-EmmyLua/issues/222)

**Comment by modo-lv** [comment]
I would like to add my vote for LDoc support. LDoc can automatically generate full-fledged documentation from the comments and is widely used.

**Comment by Novakov** [comment]
Another 👍 from me. I found out about EmmyLua in IntelliJ and VS Code and I really, really like it. I would like also to be able to build some documentation using ldoc but with annotations being incompatible I'm stuck with choice between getting code completion or getting docs, sadly I have to choose none.

I'm not heavy Lua user and I don't know anything about IntelliJ details, however I would like to offer my help for example (I guess only testing and providing sample files are the only thing I'm capable for doing for now).

**Comment by redcatbear** [comment]
I agree that standard LDoc support would be a valuable improvement, especially because existing linters and documentation renderers all expect LDoc.

**Comment by redcatbear** [comment]
@tangzx, could you give us a hint please, where we can see the part of the build that turns the provided BNF into the generated Java?

**Comment by CppCXY** [comment]
install idea grammar kit, and right click and select generate parse code.
but emmylua perhaps won't choose compatible with LDoc.
the emmylua doc has been accepted by many, If you are really interested in this, you can choose to fork this project and publish an independent plugin
